File: ios/Plugin/ZendeskChatPlugin.swift
import Foundation
import Capacitor
import ZDCChat

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(ZendeskChatPlugin)
public class ZendeskChatPlugin: CAPPlugin {
    private let implementation = ZendeskChat()

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": implementation.echo(value)
        ])
    }

    @objc func initialize(_ call: CAPPluginCall) {
        DispatchQueue.main.async {
            let accountKey = self.getConfigValue("accountKey") as? String
            ZDCChat.initialize(withAccountKey: accountKey)
        }
    }

    @objc func setVisitorInfo(_ call: CAPPluginCall) {
        DispatchQueue.main.async {
            let name = call.getString("name")
            let email = call.getString("email")
            let phone = call.getString("phoneNumber")
            ZDCChat.updateVisitor { user in
                user?.name = name
                user?.email = email
                user?.phone = phone
            }
        }
    }

    @objc func open(_ call: CAPPluginCall) {
        DispatchQueue.main.async {
            let department = call.getString("department") ?? "testing"
            let tags = call.getArray("tags", String.self) ?? [];
            ZDCChat.start(in: self.bridge?.viewController?.navigationController, withConfig: {config in
                config?.department = department
                config?.tags = tags
            })
        }
    }
}

File: android/src/main/java/com/mycompany/plugins/example/ZendeskChatPlugin.java
package com.mycompany.plugins.example;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;
import com.getcapacitor.JSArray;
import com.getcapacitor.NativePlugin;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.zopim.android.sdk.api.ZopimChat;
import com.zopim.android.sdk.model.VisitorInfo;
import com.zopim.android.sdk.prechat.ZopimChatActivity;

import org.json.JSONException;

@CapacitorPlugin(name = "ZendeskChat")
public class ZendeskChatPlugin extends Plugin {

    private ZendeskChat implementation = new ZendeskChat();

    @PluginMethod
    public void echo(PluginCall call) {
        String value = call.getString("value");

        JSObject ret = new JSObject();
        ret.put("value", implementation.echo(value));
        call.resolve(ret);
    }

  @PluginMethod()
  public void initialize(PluginCall call) {
    String accountKey = String.valueOf(this.getConfigValue("accountKey"));
    ZopimChat.init(accountKey);
  }

  @PluginMethod()
  public void setVisitorInfo(PluginCall call) {
    String name = call.getString("name");
    String email = call.getString("email");
    String phoneNumber = call.getString("phoneNumber");

    VisitorInfo visitorData = new VisitorInfo.Builder()
      .name(name)
      .email(email)
      .phoneNumber(phoneNumber)
      .build();

    ZopimChat.setVisitorInfo(visitorData);
  }

  @PluginMethod()
  public void open(PluginCall call) {
    String department = call.getString("department");
    JSArray tags = call.getArray("tags", new JSArray());

    String[] tagsParameters = null;
    try {
      tagsParameters = tags.<String>toList().toArray(new String[0]);
    } catch (JSONException e) {
      e.printStackTrace();
    }

    AppCompatActivity activity = getActivity();

    ZopimChat.SessionConfig config = new ZopimChat.SessionConfig()
      .department(department)
      .tags(tagsParameters);

    ZopimChatActivity.startActivity(activity, config);
  }
}
